File: app/src/main/java/com/dhis2/data/server/ServerComponent.java
package com.dhis2.data.server;

import android.support.annotation.NonNull;

import com.dhis2.data.dagger.PerServer;
import com.dhis2.data.user.UserComponent;
import com.dhis2.data.user.UserModule;

import dagger.Subcomponent;

@PerServer
@Subcomponent(modules = ServerModule.class)
public interface ServerComponent {

    @NonNull
    UserManager userManager();

    @NonNull
    UserComponent plus(@NonNull UserModule userModule);
}

File: app/src/main/java/org/dhis2/data/server/ServerComponent.kt
package org.dhis2.data.server

import org.dhis2.data.dagger.PerServer
import org.dhis2.data.user.UserComponent
import org.dhis2.data.user.UserModule

import dagger.Subcomponent

@PerServer
@Subcomponent(modules = [ServerModule::class])
interface ServerComponent {

    fun userManager(): UserManager

    operator fun plus(userModule: UserModule): UserComponent
}
